### PR TITLE
Add lite dev mode so GCP auth is not required

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: developer_guide.md [hunk 1]
```diff
@@ -143,6 +144,14 @@ Start the flask webserver locally at localhost:8080
 ./run_server.sh
 ```
 
+If you don't have DataCommons GCP permissions, can run
```
Comment (beets) [inline developer_guide.md:147]:
```suggestion
If you don't have DataCommons GCP permissions, run
```

Comment (beets) [inline developer_guide.md:147]: Should we move the flask section up (before prerequisites) and change it to "simple / lite" instead of not recommended?

**Review by beets — APPROVED** [review]
thanks for this bo!

should we also avoid updating the mixer submodule unless it's release related or required dependency change?

File: developer_guide.md (hunk 1 above)
Comment (shifucun) [reply developer_guide.md]: Done

Comment (shifucun) [reply developer_guide.md]: Done

**Comment by shifucun** [comment]
> thanks for this bo!
> 
> should we also avoid updating the mixer submodule unless it's release related or required dependency change?

oops, that's not intended. Reverted. Thanks for the catch!

**Pull request merged**